docs/audio_driver.md

```diff
@@ -0,0 +1,122 @@
+# Audio Driver
+
+The [Audio feature](feature_audio.md) breaks the hardware specifics out into separate, exchangeable driver units, with a common interface to the audio-"core" - which itself handles playing songs and notes while tracking their progress in an internal state, initializing/starting/stopping the driver as needed.
+
+Not all MCUs support every available driver, either the platform-support is not there (yet?) or the MCU simply does not have the required hardware peripheral.
+
+
+## AVR
+
+Boards build around an Atmega32U4 can use two sets of PWM capable pins, each driving a separate speaker.
+The possible configurations are:
+
+|              | Timer3      | Timer1       |
+| one speaker  | C4,C5 or C6 |              |
+| one speaker  |             | B4, B5 or B7 |
+| two speakers | C4,C5 or C6 | B4, B5 or B7 |
+
+Currently there is only one/default driver for AVR based boards, which is automatically configured to:
+```make
+AUDIO_DRIVER = pwm_hardware
+```
+
+
+## ARM
+
+For Arm based boards, QMK depends on ChibiOS - hence any MCU supported by the later is likely usable, as long as certain hardware peripherals are available.
+
+Supported wiring configurations, with their ChibiOS/MCU peripheral requirement are listed below;
+piezo speakers are marked with :one: for the first/primary and :two: for the secondary.
+
+  | driver       | GPTD6                            | GPTD7                  | GPTD8 | PWMD1 [^1]      |
+  |              | Tim6                             | Tim7                   | Tim8  | Tim1_Ch1        |
+  |--------------|----------------------------------|------------------------|-------|-----------------|
+  | dac_basic    | A4+DACD1 = :one:                 | A5+DACD2 = :one:       | state |                 |
+  |              | A4+DACD1 = :one: + Gnd           | A5+DACD2 = :two: + Gnd | state |                 |
+  |              | A4+DACD1 = :two: + Gnd           | A5+DACD2 = :one: + Gnd | state |                 |
+  |              | A4+DACD1 = :one: + Gnd           |                        | state |                 |
+  |              |                                  | A5+DACD2 = :one: + Gnd | state |                 |
+  | dac_additive | A4+DACD1 = :one: + Gnd           |                        |       |                 |
+  |              | A5+DACD2 = :one: + Gnd           |                        |       |                 |
+  |              | A4+DACD1 + A5+DACD2 = :one: [^2] |                        |       |                 |
+  | pwm_software | state-update                     |                        |       | any = :one:     |
+  | pwm hardware | state-update                     |                        |       | A8 = :one: [^3] |
```

Comment:
Markdown doesn't support multiple table header rows, you have to line break the headings manually.

GitHub-Flavored Markdown also doesn't support the `^` syntax for superscript; you have to use the `<sup>` HTML tag.

```suggestion
  | driver       | GPTD6<br>Tim6                            | GPTD7<br>Tim7          | GPTD8<br>Tim8 | PWMD1<sup>1</sup><br>Tim1_Ch1 |
  |--------------|------------------------------------------|------------------------|---------------|-------------------------------|
  | dac_basic    | A4+DACD1 = :one:                         | A5+DACD2 = :one:       | state         |                               |
  |              | A4+DACD1 = :one: + Gnd                   | A5+DACD2 = :two: + Gnd | state         |                               |
  |              | A4+DACD1 = :two: + Gnd                   | A5+DACD2 = :one: + Gnd | state         |                               |
  |              | A4+DACD1 = :one: + Gnd                   |                        | state         |                               |
  |              |                                          | A5+DACD2 = :one: + Gnd | state         |                               |
  | dac_additive | A4+DACD1 = :one: + Gnd                   |                        |               |                               |
  |              | A5+DACD2 = :one: + Gnd                   |                        |               |                               |
  |              | A4+DACD1 + A5+DACD2 = :one: <sup>2</sup> |                        |               |                               |
  | pwm_software | state-update                             |                        |               | any = :one:                   |
  | pwm hardware | state-update                             |                        |               | A8 = :one: <sup>3</sup>       |
```